src/components/featured-products-section/featured-products-section.tsx

```diff
@@ -18,50 +18,45 @@ const getFeaturedProducts = async (
 ): Promise<FeaturedProductsData | null> => {
     const api = getEcomApi();
 
-    let category: collections.Collection;
+    let category: collections.Collection | undefined;
     try {
-        category = await api.getCategoryBySlug(categorySlug);
+        category =
+            (await api.getCategoryBySlug(categorySlug)) ||
+            (await api.getCategoryBySlug('all-products'));
     } catch (error) {
-        try {
+        if (isEcomSDKError(error) && error.details.applicationError.code === 404) {
             category = await api.getCategoryBySlug('all-products');
-        } catch (error) {
-            return null;
+        } else {
+            throw error;
         }
     }
+    if (!category) throw new Error("Category 'all-products' not found");
```

Comment:
How is it possible for it to be `undefined`?

Reply:
I don't know how. 
<img width="1140" alt="Screenshot 2024-09-21 at 07 22 02" src="https://github.com/user-attachments/assets/d9aa3f76-2b02-4639-995b-8c1dc4b67f29">